### PR TITLE
cleanup agent execute method

### PR DESCRIPTION
**Pull request by jsvd**
work towards decoupling the agent from the pipeline by moving config handling and pipeline setup to separate methods

### CODE REVIEW

File: lib/logstash/agent.rb [hunk 1]
```diff
@@ -127,36 +98,45 @@ def execute
...
     report I18n.t("oops", :error => e)
     report e.backtrace if @logger.debug? || $DEBUGLIST.include?("stacktrace")
     return 1
   ensure
+    @logger.unsubscribe(stdout_logs) if show_startup_errors
     @log_fd.close if @log_fd
     Stud::untrap("INT", trap_id) unless trap_id.nil?
   end # def execute
 
+  def format_config config_path, config_string
```
Comment (ph) [inline lib/logstash/agent.rb]: Would add parentheses to this line.
def format_config(config_path, config_string)

File: lib/logstash/agent.rb [hunk 2]
```diff
@@ -85,41 +84,13 @@ def execute
...
-    rescue LoadError => e
-      fail("Configuration problem.")
-    end
+    configure
 
     # Make SIGINT shutdown the pipeline.
     trap_id = Stud::trap("INT") do
       @logger.warn(I18n.t("logstash.agent.interrupted"))
-      pipeline.shutdown
+      @pipeline.shutdown if @pipeline
```
Comment (jordansissel) [inline lib/logstash/agent.rb]: What's the reason in having `@pipeline` be an ivar?

Comment (jsvd) [inline lib/logstash/agent.rb]: no reason

**Comment by jordansissel** [comment]
Code LGTM pretty much good loveliness and warmth.  I didn't test it though.

**Comment by jsvd** [comment]
rebased since it had deviated from master, another review welcome please.

**Comment by jsvd** [comment]
this has been done in other prs, closing.

**Pull request closed without merging**